src/ring_theory/chain_of_divisors.lean

```diff
@@ -195,13 +196,38 @@ lemma is_prime_pow_of_has_chain {q : associates M} {n : ℕ} (hn : n ≠ 0)
 
 end divisor_chain
 
-variables {N : Type*} [cancel_comm_monoid_with_zero N] [unique_factorization_monoid N]
-  [decidable_eq (associates M)] [unique_factorization_monoid M]
+variables {N : Type*} [cancel_comm_monoid_with_zero N]
+
+lemma map_is_unit_of_monotone_is_unit {m u : associates M} {n : associates N}
```

Comment:
I don't really understand the name, especially the `monotone` part. How about something like:
```suggestion
lemma is_unit.factor_order_iso_map {m u : associates M} {n : associates N}
```